src/jit/codegenxarch.cpp

```diff
@@ -5279,86 +5323,156 @@ void CodeGen::genConsumePutStructArgStk(
 }
 #endif // FEATURE_UNIX_AMD64_STRUCT_PASSING
 
-void CodeGen::genConsumeBlockOp(GenTreeBlkOp* blkNode, regNumber dstReg, regNumber srcReg, regNumber sizeReg)
+//------------------------------------------------------------------------
+// genConsumeBlockSize: Ensure that the block size is in the given register
+//
+// Arguments:
+//    blkNode - The block node
+//    sizeReg - The register into which the block's size should go
+//
+
+void CodeGen::genConsumeBlockSize(GenTreeBlk* blkNode, regNumber sizeReg)
+{
+    unsigned blockSize = blkNode->Size();
+    if (sizeReg != REG_NA)
```

Comment:
Actually, I think there may be a bug here - in the `GT_STORE_DYN_BLK` case, it should probably be `genConsumeRegAndCopy(blkNode->AsDynBlk()->gtDynamicSize, sizeReg)`. This probably hasn't been hit because 1) the dynamic size is not common and 2) it is probably usually already in the right register.

---
In reply to: [77254555](https://github.com/dotnet/coreclr/pull/7027#discussion_r77254555) [](ancestors = 77254555)